pandas/tests/io/test_stata.py

```diff
@@ -1608,3 +1610,15 @@ def test_strl_latin1(self):
                     val = gso.split(b'\x00')[-2]
                     size = gso[gso.find(b'\x82') + 1]
                     assert len(val) == size - 1
+
+    def test_encoding_latin1_118(self):
+        # GH 25960
+        msg = """
+One or more strings in the dta file could not be decoded using utf-8, and
+so the fallback encoding of latin-1 is being used.  This can happen when a file
+has been incorrectly encoded by Stata or some other software. You should verify
+the string values returned are correct."""
+        with pytest.warns(UnicodeWarning, match=msg):
+            encoded = read_stata(self.dta_encoding_118)
```

Comment:
Use `tm.assert_produces_warning`.  Our implementation also checks stacklevel.

Reply:
actually this already checks